onnxconverter_common/auto_mixed_precision_model_path.py

```diff
@@ -0,0 +1,250 @@
+
+"""
+
+import copy
+import numpy as np
+import onnxruntime as ort
+import onnx
+import os
+import uuid
+from onnxconverter_common import float16
+from onnx import shape_inference
+from .auto_mixed_precision import SegmentList
+
+
+def auto_convert_mixed_precision_model_path(source_model_path, input_feed,
+                                            target_model_path, location=None,
+                                            validate_fn=None, rtol=None, atol=None,
+                                            keep_io_types=False, providers=None):
+    """
+    Automatically converts a model to mixed precision, excluding the minimum number of nodes required to
+    ensure valudate_fn returns True and/or results are equal according to rtol/atol
+    this version support model_path as input, which the model could > 2GB
+    """
+
+    if not isinstance(source_model_path, str):
+        raise TypeError('auto_convert_mixed_precision_model_path only accepts model Path (String),'
+                        'you can use auto_convert_mixed_precision for the ModelProto.')
+
+    if rtol is None and atol is not None:
+        rtol = 1e-5
+
+    if atol is None and rtol is not None:
+        atol = 1e-8
+
+    if rtol is None and validate_fn is None:
+        raise ValueError("Argument `validate_fn` and `rtol` cannot both be `None`.")
+
+    if location is None:
+        location = "fp16_tensor.data"
+
+    tmp_model32_path, tmp_model32_tensor_name = generate_temp_filename(target_model_path)
+
+    kwargs = {
+        "tmp_model32_path": tmp_model32_path,
+        "tmp_model32_tensor_name": tmp_model32_tensor_name,
+        "source_model_path": source_model_path,
+        "input_feed": input_feed,
+        "target_model_path": target_model_path,
+        "location": location,
+        "validate_fn": validate_fn,
+        "rtol": rtol,
+        "atol": atol,
+        "keep_io_types": keep_io_types,
+        "providers": providers
+        }
+
+    print("copy source model to temp folder, change to external data, then check.")
+    model_32, output_32 = _copy_fp32_model(**kwargs)
+
+    print("convert to initial fp16 model, then check.")
+    node_names = [n.name for n in model_32.graph.node if n.op_type not in ["Loop", "If", "Scan"]]
+    kwargs["model_32"] = model_32
+    kwargs["res1"] = output_32
+    kwargs["node_block_list"] = node_names
+    kwargs["return_model"] = False
+    _convert_and_check_inference_result(**kwargs)
+
+    print("Sanity checks passed. Starting autoconvert.")
+    final_nodes = _try_to_convert_to_valid_fp16_model(**kwargs)
+
+    print(".final convert.")
+    kwargs["node_block_list"] = final_nodes
+    kwargs["return_model"] = True
+    valid, model = _convert_and_check_inference_result(**kwargs)
+    if not valid:
+        raise ValueError("validation failed for final fp16 model")
+    print("Final model validated successfully.")
+
+    _clean_output_folder(**kwargs)
```

Comment:
Given the file size of a model would be great, considering a try block before this so that the extra model file we generated could be cleaned even any of previous steps failed.

Thoughts?